spec/models/retirement_manager_spec.rb

```diff
@@ -0,0 +1,33 @@
+describe RetirementManager do
+  describe "#check" do
+    let(:ems) { FactoryGirl.create(:ext_management_system) }
+    let(:zone) { FactoryGirl.create(:zone, :name => "api_zone", :ext_management_systems => [ems]) }
+    let(:my_server) { FactoryGirl.create(:miq_server, :zone => zone) }
+
+    it "check runs" do
+      allow(MiqServer).to receive(:my_server).and_return(my_server)
+      expect(RetirementManager).to receive(:check)
+
+      RetirementManager.check
+    end
+
+    it "with retirement date, runs retirement checks" do
+      FactoryGirl.create(:load_balancer, :retires_on => Time.zone.today + 1.day, :ems_id => ems.id)
+      FactoryGirl.create(:load_balancer)
+      FactoryGirl.create(:orchestration_stack, :retires_on => Time.zone.today + 1.day, :ems_id => ems.id)
+      FactoryGirl.create(:orchestration_stack)
+      FactoryGirl.create(:vm, :retires_on => Time.zone.today + 1.day, :ems_id => ems.id)
+      FactoryGirl.create(:vm)
+      FactoryGirl.create(:service, :retires_on => Time.zone.today + 1.day)
+      FactoryGirl.create(:service, :retired => true)
+      allow(MiqServer).to receive(:my_server).and_return(my_server)
+
+      expect_any_instance_of(LoadBalancer).to receive(:retirement_check).once.and_return(true)
```

Comment:
I think these checks should expect a specific instance to receive the `retirement_check`.  Also, the other instances should expect not to receive `retirement_check`, right?